custom/gcc_sangath/rules/custom_actions.py

```diff
@@ -0,0 +1,95 @@
+from corehq.apps.data_interfaces.models import (
+    AUTO_UPDATE_XMLNS,
+    CaseRuleActionResult,
+)
+from corehq.apps.hqcase.utils import update_case
+from corehq.form_processor.models import CommCareCase, CommCareCaseIndex
+from custom.gcc_sangath.const import (
+    DATE_OF_PEER_REVIEW_CASE_PROP,
+    MEAN_GENERAL_SKILLS_SCORE_CASE_PROP,
+    MEAN_TREATMENT_SPECIFIC_SCORE_CASE_PROP,
+    NEEDS_AGGREGATION_CASE_PROP,
+    NEEDS_AGGREGATION_NO_VALUE,
+    SESSION_CASE_TYPE,
+    SESSION_RATING_CASE_PROP,
+)
+
+
+def sanitize_session_peer_rating(session_case, rule):
+    """
+    For any session case
+    sanitize the peer_rating metrics based on child cases of type peer_rating
+    """
+    if session_case.type != SESSION_CASE_TYPE:
+        return CaseRuleActionResult()
+
+    peer_rating_cases = _get_peer_rating_cases(session_case)
+    num_updates = 0
+
+    if peer_rating_cases:
+        case_updates = _get_case_updates(peer_rating_cases)
+        (submission, cases) = update_case(
+            session_case.domain,
+            session_case.case_id,
+            case_properties=case_updates,
+            xmlns=AUTO_UPDATE_XMLNS,
+            device_id=__name__ + ".sanitize_session_peer_rating",
+        )
+        num_updates = 1
+        rule.log_submission(submission.form_id)
+
+    return CaseRuleActionResult(
+        num_updates=num_updates,
+    )
+
+
+def _get_peer_rating_cases(session_case):
+    case_ids = CommCareCaseIndex.objects.get_extension_case_ids(session_case.domain, [session_case.case_id])
```

Comment:
Would it be worth also filtering on case type? Just in case?

Reply:
oh yes absolutely, i will add that.

Reply:
https://github.com/dimagi/commcare-hq/pull/31119/commits/ff53334a8dc36b2f2700f8c395175cdecb12084e